File: convert-to-jpg.mjs
import sharp from 'sharp';
import { readdirSync, existsSync } from 'fs';
import { join, basename, extname } from 'path';

const IMG_DIR = './img';

const dirs = readdirSync(IMG_DIR, { withFileTypes: true })
  .filter(d => d.isDirectory())
  .map(d => join(IMG_DIR, d.name));

let total = 0;

for (const dir of dirs) {
  const files = readdirSync(dir).filter(f => f.endsWith('.webp'));
  for (const file of files) {
    const src  = join(dir, file);
    const dest = join(dir, basename(file, '.webp') + '.jpg');
    if (existsSync(dest)) { console.log(`SKIP  ${dest}`); continue; }
    await sharp(src).jpeg({ quality: 88, progressive: true }).toFile(dest);
    console.log(`OK    ${dest}`);
    total++;
  }
}

console.log(`\nFertig: ${total} JPG-Dateien erstellt.`);

File: convert-to-jpg.py
from PIL import Image
from pathlib import Path

img_dir = Path('./img')
total = 0

for webp in sorted(img_dir.rglob('*.webp')):
    jpg = webp.with_suffix('.jpg')
    if jpg.exists():
        print(f'SKIP  {jpg}')
        continue
    img = Image.open(webp).convert('RGB')
    img.save(jpg, 'JPEG', quality=88, optimize=True, progressive=True)
    print(f'OK    {jpg}')
    total += 1

print(f'\nFertig: {total} JPG-Dateien erstellt.')
